plugins/by-name/roslyn/default.nix

```diff
@@ -0,0 +1,15 @@
+{ lib, pkgs, ... }:
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "roslyn";
+  packPathName = "roslyn.nvim";
+  package = "roslyn-nvim";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  extraConfig = {
+    plugins.roslyn.settings.exe = lib.mkDefault "Microsoft.CodeAnalysis.LanguageServer";
+  };
+
+  # TODO: Figure out how to add this package, as it's the source of the `Microsoft.CodeAnalysis.LanguageServer` command, which is required for the LSP to function
+  # home.packages = [ pkgs.roslyn-ls ];
```

Comment:
I'm unfamiliar with this LS, so I don't know if it is better to contribute to nvim-lspconfig (so that our `plugins.lsp.servers` options can be used) or to assume all configuration will be done by this plugin separately from nvim-lspconfig.

If the LS can't really be configured by nvim-lspconfig, the next best thing would be to use our new `dependencies` options.

To do this, you would edit `modules/dependencies.nix`, and add `"roslyn-ls"` to the list. Then you would edit `extraConfig` here, and define `dependencies.roslyn-ls.enable = lib.mkDefault true`.